nomenclature/codelist.py

```diff
@@ -388,6 +389,49 @@ def codelist_repr(self, json_serialized=False) -> Dict:
 
         return nice_dict
 
+    def filter(self, **kwargs) -> "CodeList":
+        """Filter a CodeList by any attribute-value pairs.
+
+        Parameters
+        ----------
+        **kwargs
+            a variable number of keyword arguments representing
+            attribute-value pairs.
+
+        Returns
+        -------
+        Instance of 'CodeList'
+            A new CodeList object with only the Code objects that
+            match all the attribute-value pairs.
```

Comment:
This can be made shorter...
```suggestion
        CodeList
```